specification/network/resource-manager/Microsoft.Network/stable/2024-01-01/webapplicationfirewall.json

```diff
@@ -899,8 +899,20 @@
         "selectorMatchOperator",
         "selector"
       ],
-      "description": "Allow to exclude some variable satisfy the condition for the WAF check.",
+      "description": "Allows exclusion of a variable or the whole request from WAF check when the exclusion condition is satisfied.",
       "properties": {
+        "exclude": {
+          "type": "string",
+          "enum": [
+            "MatchVariable",
```

Comment:
Since this is optional parameter, what would be default value if not passed for existing clients?

Reply:
Exclude's default value is set to "MatchVariable" in the RP code. I can add the default here as well.

Quick question on default + required behavior: If we add exclude as requried and set it's default value, will the default replacement happen before or after the required check? To elaborate, if Cx doesn't send the exclude field in the request, will it be added with the default value and then the validation for required fields take place? Or does the required fields validation done at the beginning?

I wasn't able to get a clear idea from going through OpenAPI's docs, would appreciate your insights.

Reply:
default value is applicable for optional variables.. so either you can make variable optional and have default value on service side or make this parameter mandatory for new API version and get approval for cross-version breaking changes. ARM wiki: https://eng.ms/docs/products/arm/api_contracts/guidelines/openapi OAPI022

Reply:
Got it, thanks! Will keep it optional with a default value.